src/groupchat.c

```diff
@@ -301,6 +352,65 @@ static void groupchat_onGroupTitleChange(ToxWindow *self, Tox *m, uint32_t group
     write_to_log(tmp_event, nick, ctx->log, true);
 }
 
+/* Puts `(NameListEntry *)`s in `entries` for each matched peer, up to a maximum
+ * of `maxpeers`.
+ * Maches each peer whose name or pubkey begins with `prefix`.
+ * If `prefix` is exactly the pubkey of a peer, matches only that peer.
+ * return number of entries placed in `entries`.
+ */
+uint32_t get_name_list_entries_by_prefix(uint32_t groupnumber, const char *prefix, NameListEntry **entries,
+        uint32_t maxpeers)
+{
+    GroupChat *chat = &groupchats[groupnumber];
+
+    if (!chat->active) {
+        return 0;
+    }
+
+    const int len = strlen(prefix);
```

Comment:
You may want to use size_t here, or cast to `(int)` explicitly if you mean a conversion.